examples/chat/client/tasks.nim

```diff
@@ -729,3 +729,120 @@ proc publishWakuChat2*(message: string) {.task(kind=no_rts, stoppable=false).} =
+    asyncSpawn chanSendToHost.send(eventNotLoggedInEnc.safe)
+    return
+
+
+  let tokens = status.getCustomTokens()
+  if tokens.isErr:
+    let
+      event = GetCustomTokensEvent(error: tokens.error)
+      eventEnc = event.encode
+      task = taskArg.taskName
+
+    trace "task sent errored event to host", event=eventEnc, task
+    asyncSpawn chanSendToHost.send(eventEnc.safe)
+    return
+
+  let
+    event = GetCustomTokensEvent(tokens: tokens.get,
+      timestamp: getTime().toUnix)
+    eventEnc = event.encode
+    task = taskArg.taskName
+
+  trace "task sent event to host", event=eventEnc, task
+  asyncSpawn chanSendToHost.send(eventEnc.safe)
+
+proc addCustomToken*(address: string, name: string, symbol: string, color: string, decimals: string) {.task(kind=no_rts, stoppable=false).} =
+  let timestamp = getTime().toUnix
+
+  if statusState != StatusState.loggedin:
+    let
+      eventNotLoggedIn = AddCustomTokenEvent(error: "Not logged in, " &
+        "cannot add a new custom token.",
+        timestamp: timestamp)
+      eventNotLoggedInEnc = eventNotLoggedIn.encode
+      task = taskArg.taskName
+
+    trace "task sent event to host", event=eventNotLoggedInEnc, task
+    asyncSpawn chanSendToHost.send(eventNotLoggedInEnc.safe)
+    return
+
+  try:
+    let
+      addResult = status.addCustomToken(address.parseAddress, name, symbol, color, decimals.parseUInt)
+
+    if addResult.isErr:
+      raise newException(Exception, addResult.error)
+
+    let
+      token = addResult.get
+      event = AddCustomTokenEvent(address: $token.address,
+        name: token.name, symbol: token.symbol, color: token.color, 
+        decimals: token.decimals, timestamp: timestamp)
+      eventEnc = event.encode
+      task = taskArg.taskName
+
+    trace "task sent event to host", event=eventEnc, task
+    asyncSpawn chanSendToHost.send(eventEnc.safe)
+  except Exception as e:
+    let
+      event = AddCustomTokenEvent(error: "Error adding a custom token, " &
+        "error: " & e.msg, timestamp: timestamp)
+      eventEnc = event.encode
+      task = taskArg.taskName
+    trace "task sent event with error to host", event=eventEnc, task
+    asyncSpawn chanSendToHost.send(eventEnc.safe)
+
+
+proc deleteCustomToken*(address: string) {.task(kind=no_rts, stoppable=false).} =
+  let timestamp = getTime().toUnix
+
+  if statusState != StatusState.loggedin:
+    let
+      eventNotLoggedIn = DeleteCustomTokenEvent(error: "Not logged in, " &
+        "cannot delete a custom token.",
+        timestamp: timestamp)
+      eventNotLoggedInEnc = eventNotLoggedIn.encode
+      task = taskArg.taskName
+
+    trace "task sent event to host", event=eventNotLoggedInEnc, task
+    asyncSpawn chanSendToHost.send(eventNotLoggedInEnc.safe)
+    return
+
+  try:
+    let
+      deleteResult = status.deleteCustomToken(address.parseAddress) 
+
+    if deleteResult.isErr:
+      raise newException(Exception, deleteResult.error)
```

Comment:
Actually this is never guaranteed to be caught. It's better to send an error event to the task as you've already done above.

Reply:
Question - deleting by index then means we have to add ordering to the `tokens` DB table, right?

Reply:
Hmm, yes good point. We could order by symbol maybe? What do you think?

Reply:
Added ordering by symbol. Did it yesterday actually (by address), but forgot to click the `Comment` button.

Reply:
Also - modified DeleteCustomToken to accept `index` parameter.